### PR TITLE
Allow multiple merchants - remove global config

### PR DESCRIPTION
**Pull request by SvenErikPy**
Slightly adjusted version of the work the BEKK summer students did
For each set of credentials, create a VippsApi, and use the services on it.

### CODE REVIEW

File: src/Vipps.net/VippsApi.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        {
+            return _accessTokenService;
+        }
+
+        public IVippsEpaymentService EpaymentService()
+        {
+            return _epaymentService;
+        }
+
+        public IVippsCheckoutService CheckoutService()
```
Comment (sebdak) [inline src/Vipps.net/VippsApi.cs:73]: Does this need to be a function? can the constructor do this instead?

Comment (SvenErikPy) [reply src/Vipps.net/VippsApi.cs:73]: I was thinking making them properties instead, if that's what you mean? 
There's no logic here, only reading a private field

Comment (sebdak) [reply src/Vipps.net/VippsApi.cs:73]: I misunderstood a bit. i see it is done in the constructor. I rephrase the question to: why does this need to be a callable function? cant the checkoutservice just be accessed as a public member of the class?

Comment (SvenErikPy) [reply src/Vipps.net/VippsApi.cs:73]: Yeah agreed

Comment (SvenErikPy) [reply src/Vipps.net/VippsApi.cs]: Fixed

File: README.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ var request = new InitiateSessionRequest
     }
 };
 
-var result = await CheckoutService.InitiateSession(request);
+var result = await vippsApi.CheckoutService().InitiateSession(request);
```
Comment (sebdak) [inline README.md:60]:
```suggestion
var result = await vippsApi.CheckoutService.InitiateSession(request);
```

File: README.md [hunk 3]
```diff
@@ -97,6 +97,6 @@ All response objects have a property called `RawResponse` that contains the resp
 **`RawResponse` example:**
 
 ```c#
-var response = checkoutService.InitiateSession(initiateSessionRequest);
+var response = vippsApi.CheckoutService().InitiateSession(initiateSessionRequest);
```
Comment (sebdak) [inline README.md:100]:
```suggestion
var response = vippsApi.CheckoutService.InitiateSession(initiateSessionRequest);
```

File: src/Vipps.net.AspCoreDemo/Startup.cs [hunk 4]
```diff
@@ -21,7 +22,29 @@ public Startup(IConfiguration configuration)
...
+            services.AddTransient<IVippsApi, VippsApi>(
+                (sp) =>
+                    new VippsApi(vippsConfigurationOptions, null, sp.GetService<ILoggerFactory>())
+            );
+
             services.AddControllers();
+            services.AddSwaggerGen(c =>
+            {
+                c.SwaggerDoc("v1", new OpenApiInfo { Title = "Your API Name", Version = "v1" });
+            });
```
Comment (sebdak) [inline src/Vipps.net.AspCoreDemo/Startup.cs:47]: Does swagger have to be here? if so, maybe change the title from "your api name"

File: src/Vipps.net/Infrastructure/BaseServiceClient.cs [hunk 5]
```diff
@@ -91,19 +91,19 @@ CancellationToken cancellationToken
                 httpContent,
                 cancellationToken
             );
+#pragma warning disable IDE0079 // Remove unnecessary suppression
```
Comment (sebdak) [inline src/Vipps.net/Infrastructure/BaseServiceClient.cs:94]: Should the comment say why the warning is disabled?

File: src/Vipps.net/VippsApi.cs [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+            _checkoutService = new VippsCheckoutService(
+                new CheckoutServiceClient(
+                    _vippsConfigurationOptions,
+                    _vippsHttpClient,
+                    _loggerFactory
+                )
+            );
+        }
+
+        public IVippsAccessTokenService AccessTokenService
```
Comment (sebdak) [inline src/Vipps.net/VippsApi.cs:63]: Do we need to have this public? they might want to use just the token for another vipps api, but thats pretty thin use case for the SDK. what do you think?

File: README.md (hunk 2 above)
Comment (SvenErikPy) [reply README.md:60]: Yup forgot the readme. Thanks! :)

File: README.md (hunk 3 above)
Comment (SvenErikPy) [reply README.md:100]: Yeah forgot readme..Fixing

File: src/Vipps.net.AspCoreDemo/Startup.cs (hunk 4 above)
Comment (SvenErikPy) [reply src/Vipps.net.AspCoreDemo/Startup.cs:47]: Added by the summer students. It does make it easy to test, so I've left it. Yeah can change the title :)

File: src/Vipps.net/Infrastructure/BaseServiceClient.cs (hunk 5 above)
Comment (SvenErikPy) [reply src/Vipps.net/Infrastructure/BaseServiceClient.cs:94]: Yeah good idea, commenting

File: src/Vipps.net/VippsApi.cs (hunk 6 above)
Comment (SvenErikPy) [reply src/Vipps.net/VippsApi.cs]: Made internal things internal

**Review by sebdak — APPROVED** [review]

**Pull request merged**